File: 5KYU/Josephus Survivor/solution.rs
fn josephus_survivor(n: i32, k: i32) -> i32 {
    let mut r: i32 = 0;
    let mut i: i32 = 1;
     
    while i != n+1 {
        r = (r + k) % i;
        i+=1;
     }
    return r + 1;
 }

File: 5KYU/Josephus Survivor/solution.php
function josephus_survivor(int $n, int $k): int {
  $r = 0;
  $i = 1;
  
  while($i != $n+1) {
    $r = ($r + $k) % $i;
    $i++;
  }
  return $r+1;
}